substrate/frame/recovery/src/lib.rs

```diff
@@ -733,6 +741,83 @@ pub mod pallet {
 			frame_system::Pallet::<T>::dec_consumers(&who);
 			Ok(())
 		}
+
+		/// Poke / update deposits for recovery configurations and active recoveries.
+		///
+		/// This can be used by accounts to possibly lower their locked amount.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// This function checks both recovery configuration deposit and active recovery deposits
+		/// of the caller:
+		/// - If the caller has created a recovery configuration, checks and adjusts its deposit
+		/// - If the caller has initiated any active recoveries, checks and adjusts those deposits
+		///
+		/// If any deposit is updated, the difference will be reserved/unreserved from the caller's
+		/// account.
+		///
+		/// The transaction is made free if any deposit is updated and paid otherwise.
+		///
+		/// Emits `DepositPoked` if any deposit is updated.
+		/// Multiple events may be emitted in case both types of deposits are updated.
+		#[pallet::call_index(9)]
+		#[pallet::weight(T::WeightInfo::poke_deposit(T::MaxFriends::get()))]
+		pub fn poke_deposit(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			let mut deposit_updated = false;
+
+			// Check recovery configuration deposit
+			<Recoverable<T>>::try_mutate(&who, |maybe_config| -> DispatchResult {
+				let Some(config) = maybe_config.as_mut() else { return Ok(()) };
+				let old_deposit = config.deposit;
+				let new_deposit = Self::get_recovery_config_deposit(config.friends.len())?;
+
+				if old_deposit == new_deposit {
+					return Ok(());
+				}
+
+				if new_deposit > old_deposit {
+					let extra = new_deposit.saturating_sub(old_deposit);
+					T::Currency::reserve(&who, extra)?;
+				} else {
+					let excess = old_deposit.saturating_sub(new_deposit);
+					let remaining_unreserved = T::Currency::unreserve(&who, excess);
+					if !remaining_unreserved.is_zero() {
+						defensive!(
+							"Failed to unreserve full amount. (Requested, Actual)",
+							(excess, excess.saturating_sub(remaining_unreserved))
+						);
+					}
+				}
+				config.deposit = new_deposit;
+				deposit_updated = true;
+
+				Self::deposit_event(Event::<T>::DepositPoked {
+					who: who.clone(),
+					kind: DepositKind::RecoveryConfig,
+					old_deposit,
+					new_deposit,
+				});
+				Ok(())
+			})?;
+
+			// Check active recovery deposits. Get the deposit required for active recovery.
+			let new_deposit = T::RecoveryDeposit::get();
+
+			// Get all active recoveries where caller is rescuer
+			let active_recoveries: Vec<_> = <ActiveRecoveries<T>>::iter()
+				.filter(|(_, rescuer, _)| rescuer == &who)
+				.map(|(lost_account, _, _)| lost_account)
+				.collect();
```

Comment:
This is unbounded iteration and could be used to DOS the chain. 

It either has to be limited and then maybe accepting an optional offset to resume where it finished the last time, or a new extrinsic that accepts the map keys.  
Probably some other solutions possible as well, did not think long about it.

Reply:
very valid and important point. 

The transaction input can help and provide us with a key into which `ActiveRecovert` deserves a `poke`?

Reply:
Yes is also an option.